### PR TITLE
feat(dns_resolve): Properly support ipv6

### PR DESCRIPTION
**Pull request by dranikpg**

### CODE REVIEW

File: util/fibers/dns_resolve.cc [hunk 1]
```diff
@@ -151,8 +146,9 @@ void ProcessChannel(ares_channel channel, AresChannelState* state, DnsResolveCal
 
 }  // namespace
 
-error_code DnsResolve(string host, uint32_t wait_ms, char dest_ip[], ProactorBase* proactor) {
+error_code DnsResolve(string host, uint32_t wait_ms, absl::Span<char> dest_ip, ProactorBase* proactor) {
```
Comment (romange) [inline util/fibers/dns_resolve.cc:149]: I prefer keeping the type as `char dest[]` and update the comments around this function

File: util/fibers/dns_resolve.cc [hunk 2]
```diff
@@ -119,17 +119,12 @@ void DnsResolveCallback(void* ares_arg, int status, int timeouts, hostent* hoste
     return set_error();
   }
 
-  if (hostent->h_addrtype != AF_INET) {
```
Comment (romange) [inline util/fibers/dns_resolve.cc:122]: I suggest keeping the whitelist of AF_INET and AF_INET6 (there are other families that we do not support)

File: util/fibers/dns_resolve.cc (hunk 1 above)
Comment (dranikpg) [reply util/fibers/dns_resolve.cc]: Changed. But why? Span is implicitly constructible from char arrays...

**Comment by dranikpg** [comment]
I don't see any tests for this area, so it's ready

File: util/fibers/dns_resolve.cc (hunk 1 above)
Comment (romange) [reply util/fibers/dns_resolve.cc]: I consider it an awkward type for public interfaces.

File: util/fibers/dns_resolve.cc (hunk 2 above)
Comment (romange) [reply util/fibers/dns_resolve.cc:122]: ping

**Comment by romange** [comment]
@dranikpg have you checked manually that it's working?

**Comment by dranikpg** [comment]
> @dranikpg have you checked manually that it's working?

With a replica on localhost

File: util/fibers/dns_resolve.cc (hunk 2 above)
Comment (dranikpg) [reply util/fibers/dns_resolve.cc:122]: I already fixed it, the changediff is just different, reversed conditions

**Review by romange — APPROVED** [review]

**Comment by romange** [comment]
@dranikpg you need to sign, imho

**Pull request merged**